### ISSUE TITLE
Integration testing on iOS with phone authentication

### ISSUE THREAD

**Issue by jetre219** [invertase/react-native-firebase#1101]
Hi, i'm actually using phone authentication with react-native-firebase and everything work as expected except for when I want to test the complete flow of my application. In fact, firebase gives a way to disable reCaptcha with iOS for integration testing, and when I try to follow the firebase's doc, it doesn't work. To disable reCaptcha, you only have to add this line to the code:
`[FIRAuth auth].settings.appVerificationDisabledForTesting = YES;`
So I added it to my AppDelegate.m, but it doesn't disable reCaptcha.

Here are all the versions I use:
In package.json:
`"react-native": "^0.55.3",
"react-native-elements": "1.0.0-beta3",
"react-native-firebase": "^4.1.0",`

In podfile:
`  # Firebase dependencies
  pod 'Firebase/Core' , '~> 4.13.0'
  pod 'Firebase/Auth'
  pod 'Firebase/Messaging'`

And my AppDelegate.m:
```
/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

#import "AppDelegate.h"
#import "RNFirebaseNotifications.h"
#import "RNFirebaseMessaging.h"

#import <Firebase.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

@import GoogleMaps;

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
[FIRApp configure];
[RNFirebaseNotifications configure];
[FIRAuth auth].settings.appVerificationDisabledForTesting = YES;
NSURL *jsCodeLocation;

  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];

  [GMSServices provideAPIKey:@"XXXXXXXXX"];

  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"jobrivet_mobile"
                                               initialProperties:nil
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

return YES;
}

- (void)application:(UIApplication *)application didReceiveLocalNotification:(UILocalNotification *)notification {
[[RNFirebaseNotifications instance] didReceiveLocalNotification:notification];
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(nonnull NSDictionary *)userInfo
fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler{
[[RNFirebaseNotifications instance] didReceiveRemoteNotification:userInfo fetchCompletionHandler:completionHandler];
}

- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings {
[[RNFirebaseMessaging instance] didRegisterUserNotificationSettings:notificationSettings];
}

@end
```

**Comment by chrisbianca** [comment]
Are you testing on a simulator or device?
This won’t work on a simulator as it relies on a silent notification to work - notifications are not supported at all by iOS simulators unfortunately.

**Comment by jetre219** [comment]
I'm testing on a simulator but on the Google doc it says that:

> On iOS, the appVerificationDisabledForTesting setting has to be set to TRUE before calling verifyPhoneNumber. This is processed without requiring any APNs token or sending silent push notifications in the background, making it easier to test in a simulator. This also disables the reCAPTCHA fallback flow.

So, normally, when you disable app verification, firebase should not do any verification at all, either it is silent push notifications or reCaptcha.

**Comment by stereodenis** [comment]
How to verify app via silent push? I've add APNS cert to Firebase, but still get "custom URL" error

![image](https://user-images.githubusercontent.com/1406461/40235515-393a8438-5ab3-11e8-9730-15a7ec7a2be9.png)
and then
![image](https://user-images.githubusercontent.com/1406461/40235528-3d7a6658-5ab3-11e8-9a15-32f8aaf30199.png)

**Comment by chrisbianca** [comment]
@jetre219 By the looks of the history, this was only added to the iOS SDK 19 days ago: https://github.com/firebase/firebase-ios-sdk/commits/22a943708d156712a97d29dcd11ee903cb9d8494/Firebase/Auth/Source/Public/FIRAuthSettings.h and taking a look at the release notes, it looks like this was included in v5.0.0: https://firebase.google.com/support/release-notes/ios

We release RNFirebase v4.2.0 today which supports v5 of the iOS pods, so I would suggest updating and giving this a try.

I'm going to close, but please feel free to re-open if updating doesn't work.
